### ISSUE TITLE
About c values calculation

### ISSUE THREAD

**Issue by mengruwu** [kevinlu1248/pyate#36]
https://github.com/kevinlu1248/pyate/blob/65c9ce8f2cdb10084d8eb2801c1bf8a03c6fbd82/src/pyate/cvalues.py#L54-L85

Hi, I want to thank you for your hard work in the implementation of so many ATE methods. It quite helps me a lot.
I have studied the original cvalues paper and I got confused since I found there are some differences in initialization and calculation of cvalues.
In initialization, why should not both `"times_nested"` and  `"number_of_nested"` be set to zero?
In updating, why should `"times_nested"` be added to the candidate frequency and `"number_of_nested"` be added by 1?
In cvalue calculation and your implementation, the formula is \log_2(|a|) * f(a) - \frac{t(a)}{c(a)} (I used original notation in the paper) and it is different from  \log_2(|a|) *(f(a) - \frac{t(a)}{c(a)}) in the original paper.

All the above things quite confused me. Maybe I misunderstood the paper or the code. Please feel free to correct me.

**Comment by kevinlu1248** [comment]
Hi, It's great to hear that this project has helped you. I apologize for the confusion these issues have caused you so far.
> In initialization, why should not both "times_nested" and "number_of_nested" be set to zero?

I interpreted the paper as that a string is a substring of itself. Otherwise, the paper would have stated "proper substring". In fact, the paper actually states "When this triple is first created, c(b) = 1 and t(b) equals the frequency of a.", which I have followed. However, if you notice a performance improvement in assuming that the paper is referring to a proper substring, feel free to let me know and I can make any adjustments as necessary.
> In updating, why should "times_nested" be added to the candidate frequency and "number_of_nested" be added by 1?

It's the other way around: "times_nested" is updated by 1 and "number_of_nested" is updated by the candidate frequency. 

Essentially, "number_of_nested" is the number of times a term _t_ is nested in another term _b_, so to find "number_of_nested", we initialise it to one and every time a longer term candidate is found to be a superstring of _t_, we increment the "number_of_nested" property of _t_ by 1. Thus, when we reach _t_ when iterating through the terms (which is sorted from greatest number of words to least number of words), "number_of_nested" is as desired, since we checked through every larger string.

The "times_nested" property follows with similar reasoning, with the only difference being that we update with the number of times a superstring _b_ appears, since we are counting the number of times _t_ appears in another term candidate, so when superstring _b_ appears, we find _t_ as a substring of _b_ in a document.

> In cvalue calculation and your implementation, the formula is \log_2(|a|) * f(a) - \frac{t(a)}{c(a)} (I used original notation in the paper) and it is different from \log_2(|a|) *(f(a) - \frac{t(a)}{c(a)}) in the original paper.

This is actually a mistake on my part. I have fixed the error.

I hope everything made since. Feel free to contact me regarding any clarifications.
